app/main.py

```diff
@@ -1,3 +1,68 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    Variables = Union["Distance", int, float]
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (float, int)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (float, int)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Variables) -> Distance:
+        if isinstance(other, float | int):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: Variables) -> Distance:
+        if isinstance(other, (float, int)):
+            if other == 0:
+                raise ZeroDivisionError(f"{other} == 0")
```

Comment:
The error message in `ZeroDivisionError` could be more descriptive. Consider changing it to something like 'Division by zero is not allowed.'